src/solvers/dg/dg_2d_parallel.jl

```diff
@@ -110,13 +129,34 @@ function create_cache(mesh::ParallelTreeMesh{2}, equations::AbstractEquations{2}
 
   mortars = init_mortars(leaf_cell_ids, mesh, elements, dg.mortar)
 
-  # MPI setup
+  mpi_cache = init_mpi_cache(mesh, elements, mpi_interfaces, nvariables(equations), nnodes(dg))
+
+  cache = (; elements, interfaces, mpi_interfaces, boundaries, mortars,
+             mpi_cache)
+
+  # Add specialized parts of the cache required to compute the volume integral etc.
+  cache = (;cache..., create_cache(mesh, equations, dg.volume_integral, dg)...)
+  cache = (;cache..., create_cache(mesh, equations, dg.mortar)...)
+
+  return cache
+end
+
+
+function init_mpi_cache(mesh, elements, mpi_interfaces, nvars, nnodes)
+  mpi_cache = MPICache()
+
+  init_mpi_cache!(mpi_cache, mesh, elements, mpi_interfaces, nvars, nnodes)
+  return mpi_cache
+end
+
+
+function init_mpi_cache!(mpi_cache, mesh, elements, mpi_interfaces, nvars, nnodes)
   mpi_neighbor_ranks, mpi_neighbor_interfaces =
-    init_mpi_neighbor_connectivity(elements, mpi_interfaces, mesh)
+  init_mpi_neighbor_connectivity(elements, mpi_interfaces, mesh)
```

Comment:
Please restore the indentation. It was to highlight that this is not a new statement but a continuation of the statement of the previous line. Same goes for the lines below.